include/grpc/event_engine/event_engine.h

```diff
@@ -363,37 +368,26 @@ class EventEngine : public std::enable_shared_from_this<EventEngine> {
     /// \a default_port may be a non-numeric named service port, and will only
     /// be used if \a address does not already contain a port component.
     ///
-    /// When the lookup is complete, the \a on_resolve callback will be invoked
-    /// with a status indicating the success or failure of the lookup.
-    /// Implementations should pass the appropriate statuses to the callback.
-    /// For example, callbacks might expect to receive DEADLINE_EXCEEDED or
+    /// When the lookup is complete or cancelled, the \a on_resolve callback
```

Comment:
Not sure if it's assumed already, but maybe add a note that cancellation is "best effort", in the sense that a lookup will not necessarily fail if it's cancelled (for example a cancellation might race with completion of a successful lookup, in which case the lookup succeeds).

Reply:
I'm on the fence on this. Since we don't have an explicit Cancel API now, adding the note here seems a bit out of context. I added the following note to the DNSResolver class declaration:

```
  /// Note: in the gRPC implementation, cancellation (through destruction) is
  /// "best effort" as if the request has completed during the cancellation, the
  /// result might still be reported through its callback.
```

PTAL. Thanks!

Reply:
I've provided a suggestion for this above.